naga/src/front/wgsl/error.rs

```diff
@@ -300,12 +300,25 @@ pub(crate) enum Error<'a> {
     DiagnosticInvalidSeverity {
         severity_control_name_span: Span,
     },
-    DiagnosticNotYetImplemented {
+    DiagnosticDuplicateTriggeringRule {
         triggering_rule: FilterableTriggeringRule,
-        span: Span,
+        triggering_rule_spans: [Span; 2],
     },
 }
 
+impl<'a> From<ConflictingDiagnosticRuleError> for Error<'a> {
+    fn from(value: ConflictingDiagnosticRuleError) -> Self {
+        let ConflictingDiagnosticRuleError {
+            triggering_rule,
+            triggering_rule_spans,
+        } = value;
+        Self::DiagnosticDuplicateTriggeringRule {
+            triggering_rule,
+            triggering_rule_spans,
+        }
+    }
+}
+
```

Comment:
thought: I'm not sure if this conversion is worth the ambiguity we're causing in other places, but I don't think it's unreasonable.

Reply:
Moving conversation to <https://github.com/gfx-rs/wgpu/pull/6148#discussion_r1826293651>.